language/sygus.tex

```diff
@@ -2626,6 +2716,47 @@ \section{Examples}%
 and hence the solution for $f$ satisfies all input constraints.
 \end{example}
 
+
+\begin{example}[Optimization with Weights]
+Consider the following example:
+\begin{lstlisting}[language=SyGuS]
+(set-logic LIA)
+(set-feature :weights true)
+(declare-weight branches)
+(synth-fun f ((x Int)) Int
+   ((I Int) (B Bool))
+   ((I Int (0 1 x (+ I I) (! (ite B I I) :branches 1)))
+    (B Bool ((>= I I) (= I I)))))
+(declare-var x Int)
+(constraint (= (f x) (ite (= x 0) 0 x)))
+(optimize-synth ((! (_ branches f) :min)))
```

Comment:
Just to check my understanding here, could we also have written:

```smt
(optimize-synth (_ branches f) :min)
```

Reply:
Actually, no, I'll update the example to make it clearer. There is a difference between annotating terms, and annotating the overall tuple.

Reply:
I'm planning to add an additional example to show the difference when multiple terms are used in optimize-synth.

Reply:
Thanks Andy